pkg/utils/kubernetes/health/health.go

```diff
@@ -211,28 +209,36 @@ var (
 	}
 )
 
+const NodeCreationThreshold = 2 * time.Minute
+
 // CheckNode checks whether the given Node is healthy.
 // A node is considered healthy if it has a `corev1.NodeReady` condition and this condition reports
 // `corev1.ConditionTrue`.
+// if the node is
 func CheckNode(node *corev1.Node) error {
+	// recently created nodes temporarily have unhealthy conditions until
+	// the container runtime and the kubelet are completely initialized
+	if time.Now().UTC().Sub(node.CreationTimestamp.UTC()) < NodeCreationThreshold {
+		return nil
+	}
```

Comment:
Hmm, why this is handled in this way? The Shoot conditions are already respecting thresholds which are configurable with the `controllers.shootCare.conditionThresholds` field in the gardenlet component config. So `EveryNodeReady` will be `Processing` for the configured threshold. Hence I don't see a reason why we introduce such hard-coded thresholds.

Reply:
valid point. I saw these temporary unhealthy nodes during scaling up & drain operations.
I have to investigate this further. Thanks for raising the concern.

Reply:
Could not reproduce it. I removed it. Will come back to it when I see it again.